modules/images/webp-uploads/load.php

```diff
@@ -623,45 +611,30 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 		return $image;
 	}
 
-	$basename = wp_basename( $metadata['file'] );
-	$urls     = $matches[0];
-	foreach ( $urls as $url ) {
-		$src_filename = wp_basename( $url );
-
-		// Replace the full size image if present.
-		if ( isset( $metadata['sources'][ $target_mime ]['file'] ) && strpos( $url, $basename ) !== false ) {
-			$image = str_replace( $src_filename, $metadata['sources'][ $target_mime ]['file'], $image );
-			continue;
-		}
-
-		if ( empty( $metadata['sizes'] ) ) {
-			continue;
-		}
-
-		$extension = wp_check_filetype( $src_filename );
-		// Extension was not set properly no action possible or extension is already in the expected mime.
-		if ( empty( $extension['type'] ) || $extension['type'] === $target_mime ) {
-			continue;
+	// Replace the full size image if present.
+	if ( isset( $metadata['sources'][ $target_mime ]['file'] ) ) {
+		$basename = wp_basename( $metadata['file'] );
+		if ( $basename !== $metadata['sources'][ $target_mime ]['file'] ) {
+			$image = str_replace(
+				$basename,
+				$metadata['sources'][ $target_mime ]['file'],
+				$image
+			);
 		}
+	}
 
-		// Find the appropriate size for the provided URL.
-		foreach ( $metadata['sizes'] as $name => $size_data ) {
-			// Not the size we are looking for.
-			if ( empty( $size_data['file'] ) || $src_filename !== $size_data['file'] ) {
-				continue;
-			}
-
-			if ( empty( $size_data['sources'][ $target_mime ]['file'] ) ) {
-				continue;
-			}
-
-			// This is the same as the file we want to replace nothing to do here.
-			if ( $size_data['sources'][ $target_mime ]['file'] === $src_filename ) {
-				continue;
-			}
-
-			$image = str_replace( $src_filename, $size_data['sources'][ $target_mime ]['file'], $image );
-			break;
+	// Replace sub sizes for the image if present.
+	foreach ( $metadata['sizes'] as $name => $size_data ) {
+		if (
+			! empty( $size_data['file'] ) &&
+			! empty( $size_data['sources'][ $target_mime ]['file'] ) &&
+			$size_data['file'] !== $size_data['sources'][ $target_mime ]['file']
+		) {
```

Comment:
While technically this is the same group of conditionals as before I would argue that this way it's harder to follow, instead 1 line condition moving on makes for a more natural and simple flow.

Reply:
I would agree with it if this condition is written on a single line. IMO having every check on its own line within one condition has the same effect as if we would add three separate conditions for every check. However, if more people think that it's hard to follow, I don't mind to update it to have separate conditions.

Reply:
I think this could be considered a personal preference, but my take here is that I would agree with @mitogh the previous way was easier to follow. I also prefer the `if not x, then bail early` approach better than `if x, run the logic`, as it avoids excessive nesting. Of course it's just a single if clause here, but I think it's easier to read if the replacement remains directly within the `foreach` loop and the checks happen before as safe guards.

So to summarize my personal take:
* I prefer the way it was before.
* However, my main point would be to reverse the condition and bail early based on that, rather than only running the logic within the if clause.
* I don't have a strong preference on multiple if clauses or a single one, but I tend to agree when split in multiple it's easier to read.

Reply:
Sounds good to me then. Updated. Thanks, @mitogh and @felixarntz.